### PR TITLE
Post saved

### LINKED ISSUE #128: Clicking save button should save a park to the database, and update the store with all saved parks from the database

### LINKED ISSUE #129: When App loads, the store should update with all saved parks from the database

### PR DESCRIPTION
**Pull request by bretm9**
### Participating Group Members:

- @bretm9 

### Code Highlights:

- `apiCalls.vue`:
  - Adds `postSaved` that converts a saved park into the correct shape for the API and POSTs it to the database
  - Adds `getSaved` that gets saved parks from the API and converts them into the correct shape for our app 
- `App.vue`:
  - Adds invocation of `getSaved` in `mounted` lifecycle hook, and then commits the data to the store's `savedParks`
- `ResultsItemDetails.vue`:
  - Adds invocation of `postSaved` in `savePark` to save a park to the database when the "save" button is clicked.
  - Adds invocation of `getSaved` in `savePark` and then commits the data to the store's `savedParks`
- `ResultsListItem.vue`
  - Adds invocation of `postSaved` in `savePark` to save a park to the database when the "save" button is clicked.
  - Adds invocation of `getSaved` in `savePark` and then commits the data to the store's `savedParks`
- `store/index.js`
  - Updates `savePark` to save all parks that are returned from the API

### Have Tests Been Added?

- [x] No.
- [ ] Yes, but all tests are not passing.
- [ ] Yes, and all are passing.

### Any background context you want to provide?
N/A

### Message/Questions for reviewer:
N/A

### Issues:

- Closes #128, closes #129

### Screenshots (if appropriate):
N/A

### Tracking Consistency:

- [x] added appropriate labels
- [x] My code follows the code style of this project and has removed all unnecessary annotations
- [ ] I have added comments on my pull request, particularly in hard-to-understand areas
- [x] All new and existing tests passed
- [x] looked at PR preview to check spelling, syntax, formatting, and completion

### CODE REVIEW

**Comment by nicolegooden** [comment]
Hey Bret! Travis provided the endpoint in one of our Slack messages a little while ago.  This is the endpoint for the GET request (retrieve all saved parks):

https://boneyard-be.herokuapp.com/api/park/all/

File: src/apiCalls.js [hunk 1]
```diff
@@ -9,3 +9,29 @@ export const getSearch = (searchTerm) => {
     .then(response => response.json())
     .catch(error => console.error(error))
 }
+
+export const postSaved = (parkToSave) => {
+  const convertedPark = {
+    name: parkToSave.name,
+    formatted_address: parkToSave.formatted_address,
+    opening_hours: parkToSave.opening_hours,
```
Comment (nicolegooden) [inline src/apiCalls.js:17]: This is a great start! I believe all of these values need to be strings, though.  

Suggestions:

`opening_hours: parkToSave.opening_hours.open_now` as a string (will either be `"true"` or `"false"`)
`photo: parkToSave.photo.photo_reference` which I believe is already a string of characters
`rating: parkToSave.rating.toString( )`

File: src/components/ResultsItemDetails.vue [hunk 2]
```diff
@@ -47,8 +48,12 @@ export default {
     }
   },
   methods: {
-    savePark() {
-      this.$store.commit('savePark', this.park)
+    async savePark() {
```
Comment (nicolegooden) [inline src/components/ResultsItemDetails.vue:51]: Nice use of `async / await` here.

File: src/components/ResultsListItem.vue [hunk 3]
```diff
@@ -21,12 +21,16 @@
...
   props: ['result'],
   methods: {
-    savePark() {
-      this.$store.commit('savePark', this.result)
+    async savePark() {
+      await postSaved()
+      getSaved()
+        .then(data => {
+          this.$store.commit('saveParks', data)
+        })
```
Comment (nicolegooden) [inline src/components/ResultsListItem.vue:33]: I bet there might be a way to do this repetitive `saveParks` method with something like `Vuex actions`.  Definitely something to look into in the near future.

Comment (bretm9) [reply src/components/ResultsListItem.vue:33]: I was thinking the same thing! I googled how to do actions and got overwhelmed, but we should definitely refactor to do that later!

**Comment by bretm9** [comment]
Oh thanks @nicolegooden for mentioning that the endpoint is up!

File: src/apiCalls.js (hunk 1 above)
Comment (bretm9) [reply src/apiCalls.js:17]: Thanks for this correction! I'll add those `toString()`s tomorrow morning!

**Review by npdarrington — APPROVED** [review]
Looking good on my end with the current suggestions made already! Thanks Bret.

**Review by stacyp2006 — APPROVED** [review]
Thanks for taking the lead on this Bret!

**Pull request merged**